queries/nim/highlights.scm

```diff
@@ -0,0 +1,797 @@
+(case
+  (of_branch values:
+    (expression_list [
+      (identifier) @constant
+      (_ (identifier) @constant)
+      (_ (_ (identifier) @constant))
+      (_ (_ (_ (identifier) @constant)))
+      (_ (_ (_ (_ (identifier) @constant))))
+      (_ (_ (_ (_ (_ (identifier) @constant)))))
+      (_ (_ (_ (_ (_ (_ (identifier) @constant))))))
+    ])))
+
+; in variant objects with "case" "of"
+(variant_declaration
+  (of_branch values:
+    (expression_list [
+      (identifier) @constant
+      (_ (identifier) @constant)
+      (_ (_ (identifier) @constant))
+      (_ (_ (_ (identifier) @constant)))
+      (_ (_ (_ (_ (identifier) @constant))))
+      (_ (_ (_ (_ (_ (identifier) @constant)))))
+      (_ (_ (_ (_ (_ (_ (identifier) @constant))))))
+    ])))
+
+; enum declaration
+(enum_field_declaration
+  (symbol_declaration
+    name: [
+      (identifier) @constant
+      (accent_quoted (identifier) @constant)
+    ]))
+
+; constants/enums in array construction
+(array_construction
+  (colon_expression
+    left: [
+      (identifier) @constant
+      (_ (identifier) @constant)
+      (_ (_ (identifier) @constant))
+      (_ (_ (_ (identifier) @constant)))
+      (_ (_ (_ (_ (identifier) @constant))))
+      (_ (_ (_ (_ (_ (identifier) @constant)))))
+      (_ (_ (_ (_ (_ (_ (identifier) @constant))))))
+    ]))
+
+; constant declaration
+(const_section
+  (variable_declaration
+    (symbol_declaration_list
+      (symbol_declaration
+        name: [
+          (identifier) @constant
+          (accent_quoted (identifier) @constant)
+          (exported_symbol (identifier) @constant)
+          (exported_symbol (accent_quoted (identifier) @constant))
+        ]))))
+
+((tuple_deconstruct_declaration
+  (symbol_declaration
+    name: [
+      (identifier) @constant
+      (accent_quoted (identifier) @constant)
+      (exported_symbol (identifier) @constant)
+      (exported_symbol (accent_quoted (identifier) @constant))
+    ])) @_tuple_decons
+  (#has-ancestor? @_tuple_decons const_section))
+
+; =============================================================================
+; @constant.builtin ; built-in constant values
+
+(nil_literal) @constant.builtin
+
+; =============================================================================
+; @constant.macro   ; constants defined by the preprocessor
+; unused
+
+; =============================================================================
+; @namespace        ; modules or namespaces
+; unused
+
+; =============================================================================
+; @symbol           ; symbols or atoms
+; unused
+
+; =============================================================================
+; overrule things
+
+; left identifier in dot_expression
+(dot_expression left: [
+  (identifier) @none
+  (accent_quoted (identifier) @none)
+] (#set! "priority" 110))
+; NOTE: it can't be know what the left identifier is, so better leave it alone
```

Comment:
```suggestion
; NOTE: we can't know what the left identifier is, so better leave it alone
```

Reply:
Wait why do need priorities for these? Can't they be moved downwards

Reply:
I removed the prio here